tests/acceptance/30_custom_promise_types/custom_promise_binary.c

```diff
@@ -0,0 +1,33 @@
+#include <stdio.h>
+
+int main()
+{
+    char *line = NULL;
+    size_t len = 0;
+
+    printf("binary 0.0.1 v1 line_based\n");
+    printf("\n");
+
+    if ( getline(&line, &len, stdin) == -1 )
+    {
+      perror( "Couldn't read from stdin" );
+      return 1;
+    }
+
+    printf("operation=validate_promise\n");
+    printf("promiser=foobar\n");
+    printf("result=valid\n");
+    printf("\n");
+
+    if ( getline(&line, &len, stdin) == -1 )
+    {
+      perror( "Couldn't read from stdin" );
+      return 1;
+    }
+
+    printf("operation=evaluate_promise\n");
```

Comment:
No big deal, but spec says that repaired should have an info log message. I would change it to `kept` or add a log message:

```suggestion
    printf("log_info=The foobar was fixed\n");
    printf("operation=evaluate_promise\n");
```